ams/api/src/main/java/com/netease/arctic/ams/api/metrics/IcebergCommitReport.java

```diff
@@ -20,153 +20,230 @@
+  public static final String REMOVED_FILE_SIZE_BYTES = "removed-files-size-bytes";
+  public static final String TOTAL_FILE_SIZE_BYTES = "total-files-size-bytes";
+  public static final String ADDED_POS_DELETES = "added-positional-deletes";
+  public static final String REMOVED_POS_DELETES = "removed-positional-deletes";
+  public static final String TOTAL_POS_DELETES = "total-positional-deletes";
+  public static final String ADDED_EQ_DELETES = "added-equality-deletes";
+  public static final String REMOVED_EQ_DELETES = "removed-equality-deletes";
+  public static final String TOTAL_EQ_DELETES = "total-equality-deletes";
+
+  private final String tableName;
+  private final Long snapshotId;
+  private final Long sequenceNumber;
+  private final String operation;
+  private final Map<String, String> metadata;
+
+  private final Timer totalDuration = new Timer();
+  private final Counter attempts = new Counter();
+  private final Counter addedDataFiles = new Counter();
+  private final Counter removedDataFiles = new Counter();
+  private final Counter totalDataFiles = new Counter();
+  private final Counter addedDeleteFiles = new Counter();
+  private final Counter addedEqualityDeleteFiles = new Counter();
+  private final Counter addedPositionalDeleteFiles = new Counter();
+  private final Counter removedDeleteFiles = new Counter();
+  private final Counter removedEqualityDeleteFiles = new Counter();
+  private final Counter removedPositionalDeleteFiles = new Counter();
+  private final Counter totalDeleteFiles = new Counter();
+  private final Counter addedRecords = new Counter();
+  private final Counter removedRecords = new Counter();
+  private final Counter totalRecords = new Counter();
+  private final Counter addedFilesSizeInBytes = new Counter();
+  private final Counter removedFilesSizeInBytes = new Counter();
+  private final Counter totalFilesSizeInBytes = new Counter();
+  private final Counter addedPositionalDeletes = new Counter();
+  private final Counter removedPositionalDeletes = new Counter();
+  private final Counter totalPositionalDeletes = new Counter();
+  private final Counter addedEqualityDeletes = new Counter();
+  private final Counter removedEqualityDeletes = new Counter();
+  private final Counter totalEqualityDeletes = new Counter();
+
+  public IcebergCommitReport(
+      String tableName, Long snapshotId, Long sequenceNumber, String operation, Map<String,
+      String> metadata) {
+    this.tableName = tableName;
+    this.snapshotId = snapshotId;
+    this.sequenceNumber = sequenceNumber;
+    this.operation = operation;
+    this.metadata = metadata;
+  }
 
   @Override
-  default String name() {
+  public String name() {
     return "iceberg_commit_report";
   }
 
   @TaggedMetrics.Tag(name = TABLE_NAME)
-  String tableName();
+  public String tableName() {
+    return this.tableName;
+  }
 
   @TaggedMetrics.Tag(name = SNAPSHOT_ID)
-  long snapshotId();
+  public long snapshotId() {
```

Comment:
I don't think the Snapshot ID is a tag for the commit metric, maybe a gauge metric value?

Reply:
+1